src/atsam/Makefile

```diff
@@ -21,8 +21,14 @@ CFLAGS-$(CONFIG_MACH_SAME70) += -Ilib/same70b/include
 CFLAGS += $(CFLAGS-y) -D__$(MCU)__ -mthumb -Ilib/cmsis-core -Ilib/fast-hash
 
 CFLAGS_klipper.elf += -nostdlib -lgcc -lc_nano
-CFLAGS_klipper.elf += -T $(OUT)src/generic/armcm_link.ld
-$(OUT)klipper.elf: $(OUT)src/generic/armcm_link.ld
+
+ifeq ($(CONFIG_MACH_SAME70),y)
+    CFLAGS_klipper.elf += -T $(OUT)src/atsam/same70_link.ld
+    $(OUT)klipper.elf: $(OUT)src/atsam/same70_link.ld
+else
+    CFLAGS_klipper.elf += -T $(OUT)src/generic/armcm_link.ld
+    $(OUT)klipper.elf: $(OUT)src/generic/armcm_link.ld
+endif
```

Comment:
I'm not sure that does the right thing.  (Makefile syntax is cryptic.)

How about:
```
samlink-y := $(OUT)src/generic/armcm_link.ld
samlink-$(CONFIG_MACH_SAME70) := $(OUT)src/atsam/same70_link.ld
CFLAGS_klipper.elf += -nostdlib -lgcc -lc_nano -T $(samlink-y)
$(OUT)klipper.elf: $(samlink-y)
```

-Kevin